src/FactMgr.cpp

```diff
@@ -513,15 +513,23 @@ void
 FactMgr::makeup_new_var_facts(vector<const Fact*>& old_facts, const vector<const Fact*>& new_facts)
 {
     size_t i;
+    // We will search the old facts repeatedly. Copy the entries to a map
+    // for faster lookup.
+    FactMap old_fact_map;
```

Comment:
I believe, instead of creating a map on demand,  we can get much better speed up by storing facts in a map at the first place. This requires replacing all 'FactVec' with 'FactMap'.